src/njoy21/input/HEATR/Card1/Nplot/test/Nplot.test.cpp

```diff
@@ -1,103 +1,50 @@
-  }{
-    iRecordStream<char> iss( std::istringstream("   -22") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE_THROWS( argument::extract< HEATR::Card1::Nplot >
-		    ( iss, nendf, nin, nout ) );
-  }{
-    iRecordStream<char> iss( std::istringstream("   -23") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE_THROWS( argument::extract< HEATR::Card1::Nplot >
-		    ( iss, nendf, nin, nout ) );
-  }{
-    iRecordStream<char> iss( std::istringstream("   20") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == 20 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   -20") );
-    nin.value = -22;
-    nout.value = -23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == -20 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   50") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == 50 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   -50") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == -50 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   -50") );
-    nin.value = -22;
-    nout.value = -23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == -50 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   99") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == 99 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   -99") );
-    nin.value = -22;
-    nout.value = -23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == -99 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   /") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE( argument::extract< HEATR::Card1::Nplot >
-	     ( iss, nendf, nin, nout ).value == 0 );
-  }{
-    iRecordStream<char> iss( std::istringstream("   100") );
-    nin.value = 22;
-    nout.value = 23;
-    REQUIRE_THROWS( argument::extract< HEATR::Card1::Nplot >
-		    ( iss, nendf, nin, nout ) );
-  }{
-    iRecordStream<char> iss( std::istringstream("   -100") );
-    nin.value = -22;
-    nout.value = -23;
-    REQUIRE_THROWS( argument::extract< HEATR::Card1::Nplot >
-		    ( iss, nendf, nin, nout ) );
-  }
-}
+SCENARIO( "nplot output values",
+  "[HEATR], [Card1], [Nplot]"){
+  GIVEN( "a valid nendf tape value" ){
+    Argument< HEATR::Card1::Nendf > nendf; nendf.value = 20;
+    Argument< HEATR::Card1::Nin   > nin;   nin.value   = 21;
+    Argument< HEATR::Card1::Nout  > nout;  nout.value  = 22;
+    
+    WHEN( "nplot is within range and not equal to nendf, nin, or nout" ){
+      std::vector<int> validValues{-23, 23, 0, 42, 99, -99};
```

Comment:
Yes, I was implying that `nendf`, `nin`, and `nout` should be larger so that we can check boundary conditions. See also [Brown M&Ms: A Quick Way to Determine Code Quality](http://hiltmon.com/blog/2015/10/14/brown-mms/)